lightningd/plugin.c

```diff
@@ -1240,7 +1384,7 @@ void plugins_config(struct plugins *plugins)
 {
 	struct plugin *p;
 	list_for_each(&plugins->plugins, p, list) {
-		if (p->plugin_state == UNCONFIGURED)
+		if (p->plugin_state == NEEDS_INIT)
 			plugin_config(p);
```

Comment:
I can't add a comment directly in the function, but you don't set the `AWAITING_INIT_RESPONSE` in `plugin_config()`, after sending the init request.